Latest Intel image building error

### Describe the bug

Please see the [building log](https://github.com/deepmodeling/abacus-develop/actions/runs/9756473537/job/26926850995#step:7:3800).

### Expected behavior

_No response_

### To Reproduce

_No response_

### Environment

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

In my local machine to run the Dockerfile.intel, the detail error message is:
```
Cloning into 'abacus-develop'...
-- The CXX compiler identification is IntelLLVM 2024.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/intel/oneapi/mpi/2021.11/bin/mpiicpx - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- RapidJSON found. Headers: /usr/include
-- Found Git: /usr/bin/git (found version "2.34.1") 
-- Found git: attempting to get commit info...
-- Current commit hash: cdc307b
-- Last commit date: Wed Jul 3 23:15:11 2024 +0800
-- Found Cereal: /usr/include  
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
ELPA : We need pkg-config to get all information about the elpa library
-- Found ELPA: /usr/local/lib/libelpa_openmp.so  
-- Performing Test ELPA_VERSION_SATISFIES
-- Performing Test ELPA_VERSION_SATISFIES - Success
-- Found MPI_CXX: /opt/intel/oneapi/mpi/2021.11/bin/mpiicpx (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")  
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenMP_CXX: -fiopenmp  
-- Found OpenMP: TRUE   
-- Looking for a CUDA compiler
-- Looking for a CUDA compiler - NOTFOUND
-- Found IntelMKL: /opt/intel/oneapi/mkl/2024.2/lib/libmkl_intel_lp64.so  
CMake Error at /opt/intel/oneapi/mkl/2024.2/lib/cmake/mkl/MKLConfig.cmake:830 (add_library):
  The target name
  "MKL::/opt/intel/oneapi/mkl/2024.2/lib/libmkl_intel_lp64.so" is reserved or
  not valid for certain CMake features, such as generator expressions, and
  may result in undefined behavior.
Call Stack (most recent call first):
  /opt/libtorch/share/cmake/Caffe2/public/mkl.cmake:1 (find_package)
  /opt/libtorch/share/cmake/Caffe2/Caffe2Config.cmake:103 (include)
  /opt/libtorch/share/cmake/Torch/TorchConfig.cmake:68 (find_package)
  CMakeLists.txt:446 (find_package)


-- Found Torch: /opt/libtorch/lib/libtorch.so  
-- Populating libnpy
-- Configuring done
-- Generating done
-- Build files have been written to: /abacus-develop/build/_deps/libnpy-subbuild
[ 11%] Creating directories for 'libnpy-populate'
[ 22%] Performing download step (git clone) for 'libnpy-populate'
Cloning into 'libnpy-src'...
remote: Enumerating objects: 52, done.        
remote: Counting objects: 100% (52/52), done.        
remote: Compressing objects: 100% (35/35), done.        
remote: Total 52 (delta 14), reused 30 (delta 5), pack-reused 0        
Receiving objects: 100% (52/52), 19.82 KiB | 193.00 KiB/s, done.
Resolving deltas: 100% (14/14), done.
Already on 'master'
Your branch is up to date with 'origin/master'.
[ 33%] Performing update step for 'libnpy-populate'
HEAD is now at 471fe48 Merge pull request #38 from pauljurczak/patch-2
[ 44%] No patch step for 'libnpy-populate'
[ 55%] No configure step for 'libnpy-populate'
[ 66%] No build step for 'libnpy-populate'
[ 77%] No install step for 'libnpy-populate'
[ 88%] No test step for 'libnpy-populate'
[100%] Completed 'libnpy-populate'
[100%] Built target libnpy-populate
-- Updating submodules
Submodule 'deps/LibComm' (https://github.com/abacusmodeling/LibComm.git) registered for path 'deps/LibComm'
Submodule 'deps/LibRI' (https://github.com/abacusmodeling/LibRI.git) registered for path 'deps/LibRI'
Submodule 'deps/libpaw_interface' (https://github.com/wenfei-li/libpaw_interface) registered for path 'deps/libpaw_interface'
Cloning into '/abacus-develop/deps/LibComm'...
Cloning into '/abacus-develop/deps/LibRI'...
Cloning into '/abacus-develop/deps/libpaw_interface'...
Submodule path 'deps/LibComm': checked out 'ec984514b44480e98bd1578bcacca7a19c849724'
Submodule path 'deps/LibRI': checked out '553c91c0be1d60a86e7666f0502ef866c366c600'
Submodule path 'deps/libpaw_interface': checked out 'c211c0ab330adf3cc374f50ab3edee46b174e64c'
-- Updating submodules
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Found Libxc: version 5.1.7
-- Configuring incomplete, errors occurred!
See also "/abacus-develop/build/CMakeFiles/CMakeOutput.log".
See also "/abacus-develop/build/CMakeFiles/CMakeError.log".
The command '/bin/bash -c source /opt/intel/oneapi/setvars.sh &&     git clone https://github.com/deepmodeling/abacus-develop.git --depth 1 &&     cd abacus-develop &&     cmake -B build -DENABLE_DEEPKS=ON -DENABLE_LIBXC=ON -DENABLE_LIBRI=ON -DENABLE_RAPIDJSON=ON &&     cmake --build build -j`nproc` &&     cmake --install build &&     rm -rf build &&     abacus --version' returned a non-zero code: 1

```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -215,7 +215,7 @@ if(ENABLE_LCAO)
   if(ENABLE_FFT_TWO_CENTER)
     add_compile_definitions(USE_NEW_TWO_CENTER)
   endif()
-  
+
   if(ENABLE_PEXSI)
     find_package(PEXSI REQUIRED)
     target_link_libraries(${ABACUS_BIN_NAME} ${PEXSI_LIBRARY} ${SuperLU_DIST_LIBRARY} ${ParMETIS_LIBRARY} ${METIS_LIBRARY} pexsi)
@@ -397,13 +397,7 @@ if(MKLROOT)
   find_package(IntelMKL REQUIRED)
   add_definitions(-D__MKL)
   include_directories(${MKL_INCLUDE_DIRS} ${MKL_INCLUDE_DIRS}/fftw)
-
-  # Since libtorch will find its own MKL, the fftw part conflicts with the
-  # original one. When enable deepks, mkl will be linked within
-  # ${TORCH_LIBRARIES}.
-  if(NOT ENABLE_DEEPKS)
-    list(APPEND math_libs IntelMKL::MKL)
-  endif()
+  list(APPEND math_libs IntelMKL::MKL)
 
   if(CMAKE_CXX_COMPILER_ID MATCHES Intel)
     list(APPEND math_libs -lifcore)

--- a/cmake/FindIntelMKL.cmake
+++ b/cmake/FindIntelMKL.cmake
@@ -2,9 +2,7 @@
 # Find the native MKL headers and libraries.
 #
 #  MKL_INCLUDE_DIRS - where to find mkl.h, etc.
-#  MKL_LIBRARIES    - List of libraries when using mkl.
 #  MKL_FOUND        - True if mkl found.
-#
 
 find_path(MKL_INCLUDE_DIR mkl_service.h HINTS ${MKLROOT}/include)
 
@@ -27,11 +25,6 @@ else()
 endif()
 
 if(IntelMKL_FOUND)
-  if (ENABLE_MPI)
-    set(MKL_LIBRARIES ${MKL_INTEL} ${MKL_INTEL_THREAD} ${MKL_CORE} ${MKL_SCALAPACK} ${MKL_BLACS_INTELMPI})
-  else()
-    set(MKL_LIBRARIES ${MKL_INTEL} ${MKL_INTEL_THREAD} ${MKL_CORE})
-  endif()
   set(MKL_INCLUDE_DIRS ${MKL_INCLUDE_DIR})
 
   if(NOT TARGET IntelMKL::INTEL)

